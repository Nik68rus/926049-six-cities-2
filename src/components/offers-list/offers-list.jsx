import Card from '../card/card';

export default class OffersList extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeCard: -1
    };

    this._offerMouseEnterHandler = this._offerMouseEnterHandler.bind(this);
    this._offerMouseLeaveHandler = this._offerMouseLeaveHandler.bind(this);
  }

  render() {
    const {offers} = this.props;
    return <div className="cities__places-list places__list tabs__content">
      {
        offers.map((offer, i) => <Card key={offer.id} offer={offer} id={i} mouseEnterHandler={this._offerMouseEnterHandler} mouseLeaveHandler={this._offerMouseLeaveHandler}/>)
      }
    </div>;
  }

  _offerMouseEnterHandler(id) {
    this.setState({
      activeCard: id,
    });
  }

  _offerMouseLeaveHandler() {
    this.setState({
      activeCard: -1,
    });
  }
}

OffersList.propTypes = {
  offers: PropTypes.array.isRequired,
};

