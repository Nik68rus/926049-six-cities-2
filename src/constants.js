export const Level = {
  PREMIUM: `premium`,
};

export const Type = {
  APARTMENT: `apartment`,
  ROOM: `private room`,
};

export const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_OFFERS: `GET_OFFERS`,
  LOAD_OFFERS: `LOAD_OFFERS`,
  INIT_USER_STATE: `INIT_USER_STATE`,
  INIT_DATA_STATE: `INIT_DATA_STATE`,
  REQUIRE_AUTHORIZATION: `REQUIRE_AUTHORIZATION`,
  SIGN_IN: `SIGN_IN`,
};
