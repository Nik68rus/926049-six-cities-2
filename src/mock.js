export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Ivan`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 3,
    comment: `Beautiful place in beautifool location with good neighbors!`,
    date: Date.now(),
  },
  {
    id: 2,
    user: {
      id: 6,
      isPro: false,
      name: `Irina`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 5,
    comment: `It was awesome!`,
    date: Date.now(),
  },
  {
    id: 3,
    user: {
      id: 7,
      isPro: true,
      name: `Max`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 2,
    comment: `I would never recomend that!`,
    date: Date.now(),
  },

];
