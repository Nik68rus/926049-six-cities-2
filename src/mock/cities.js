export const CITIES = [
  {
    name: `Amsterdam`,
    location: {
      latitude: 52.38333,
      longtitude: 4.9,
      zoom: 12,
    }
  },
  {
    name: `Paris`,
    location: {
      latitude: 48.8544,
      longtitude: 2.348868,
      zoom: 10,
    }
  },
  {
    name: `Cologne`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Brussels`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Hamburg`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
];
