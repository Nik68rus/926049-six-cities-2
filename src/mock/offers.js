import {Type} from '../constants';
import {CITIES} from './cities';

export const getOffers = () => {
  let allOffers = [];
  const offersByCity = CITIES.map((city) => {
    return [
      {
        id: 0,
        city,
        title: `Beautiful & luxurious apartment at great location in ${city.name}`,
        coords: [52.3909553943508, 4.85309666406198],
        picture: `img/apartment-01.jpg`,
        type: Type.APARTMENT,
        price: 150,
        rate: 50,
        isBookmarked: true,
        isPremium: false,
      },
      {
        id: 1,
        city,
        title: `Wood and stone place in ${city.name}`,
        coords: [52.369553943508, 4.85309666406198],
        picture: `img/room.jpg`,
        type: Type.ROOM,
        price: 80,
        rate: 80,
        isBookmarked: false,
        isPremium: false,
      },
      {
        id: 2,
        city,
        title: `Canal View Prinsengracht in ${city.name}`,
        coords: [52.3909553943508, 4.929309666406198],
        picture: `img/apartment-03.jpg`,
        type: Type.APARTMENT,
        price: 120,
        rate: 90,
        isBookmarked: false,
        isPremium: false,
      },
      {
        id: 3,
        city,
        title: `Nice, cozy, warm big bed apartment in ${city.name}`,
        coords: [52.3809553943508, 4.939309666406198],
        picture: `img/apartment-02.jpg`,
        type: Type.APARTMENT,
        price: 200,
        rate: 100,
        isBookmarked: true,
        isPremium: true,
      },
    ];
  });
  offersByCity.forEach((cityOffers) => {
    cityOffers.forEach((offer) => allOffers.push(offer));
  });
  allOffers.forEach((offer, i) => {
    offer.id = i;
  });
  return allOffers;
};

// export const offers = getOffers();

export const offers = [
  {
    id: 0,
    city: CITIES[0],
    title: `Beautiful & luxurious apartment at great location in Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 1,
    city: CITIES[0],
    title: `Wood and stone place in Amsterdam`,
    coords: [52.369553943508, 4.85309666406198],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 2,
    city: CITIES[0],
    title: `Canal View Prinsengracht in Amsterdam`,
    coords: [52.3909553943508, 4.929309666406198],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 3,
    city: CITIES[0],
    title: `Nice, cozy, warm big bed apartment in Amsterdam`,
    coords: [52.3809553943508, 4.939309666406198],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 4,
    city: CITIES[1],
    title: `Beautiful & luxurious apartment at great location in Paris`,
    coords: [48.866714, 2.338152],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 5,
    city: CITIES[1],
    title: `Wood and stone place in Paris`,
    coords: [48.858179, 2.365554],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 6,
    city: CITIES[1],
    title: `Canal View Prinsengracht in Paris`,
    coords: [48.840967, 2.345730],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 7,
    city: CITIES[1],
    title: `Nice, cozy, warm big bed apartment in Paris`,
    coords: [48.852996, 2.324593],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 8,
    city: CITIES[2],
    title: `Beautiful & luxurious apartment at great location in Cologne`,
    coords: [50.930207, 6.923582],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 9,
    city: CITIES[2],
    title: `Wood and stone place in Cologne`,
    coords: [50.934906, 6.972391],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 10,
    city: CITIES[2],
    title: `Canal View Prinsengracht in Cologne`,
    coords: [50.948242, 6.951473],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 11,
    city: CITIES[3],
    title: `Nice, cozy, warm big bed apartment in Brussels`,
    coords: [50.839652, 4.334081],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 12,
    city: CITIES[3],
    title: `Beautiful & luxurious apartment at great location in Brussels`,
    coords: [50.865504, 4.368736],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 13,
    city: CITIES[4],
    title: `Wood and stone place in Hamburg`,
    coords: [53.585260, 9.984106],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 14,
    city: CITIES[4],
    title: `Canal View Prinsengracht in Hamburg`,
    coords: [53.558643, 9.967173],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 15,
    city: CITIES[4],
    title: `Nice, cozy, warm big bed apartment in Hamburg`,
    coords: [53.542315, 9.994189],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 16,
    city: CITIES[4],
    title: `Beautiful & luxurious apartment at great location in Hamburg`,
    coords: [53.542109, 9.965807],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 17,
    city: CITIES[4],
    title: `Wood and stone place in Hamburg`,
    coords: [53.550431, 9.976893],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 18,
    city: CITIES[4],
    title: `Canal View Prinsengracht in Hamburg`,
    coords: [53.545377, 9.992105],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 19,
    city: CITIES[5],
    title: `Nice, cozy, warm big bed apartment in Dusseldorf`,
    coords: [51.235248, 6.790136],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
];
