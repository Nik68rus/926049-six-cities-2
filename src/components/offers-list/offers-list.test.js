import renderer from 'react-test-renderer';
import {OffersList} from './offers-list';
import {offer} from '../../mock';
import {SortType, CardType} from '../../constants';

jest.mock(`../card/card`, () => `Card`);

it(`Offers list correctly renders after relaunch`, () => {
  const offersList = renderer
    .create(<OffersList
      cardType={CardType.CITIES}
      activeCard={0}
      offers={[offer]}
      onCardMouseEnter={jest.fn()}
      sortOrder={SortType.POPULAR}
    />)
  .toJSON();

  expect(offersList).toMatchSnapshot();
});
