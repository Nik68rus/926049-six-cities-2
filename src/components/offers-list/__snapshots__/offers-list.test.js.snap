// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers list correctly renders after relaunch 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <Card
    cardType="CITIES"
    offer={
      Object {
        "bedrooms": 3,
        "city": Object {
          "location": Object {
            "latitude": 52.370216,
            "longitude": 4.895168,
            "zoom": 10,
          },
          "name": "Amsterdam",
        },
        "coords": Array [
          52.35514938496378,
          4.673877537499948,
        ],
        "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
        "goods": Array [
          "Heating",
          "Kitchen",
          "Cable TV",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
        ],
        "host": Object {
          "avatar": "img/1.png",
          "id": 0,
          "isPro": false,
          "name": "Angelina",
        },
        "id": 0,
        "isBookmarked": false,
        "isPremium": false,
        "locationZoom": 8,
        "maxAdults": 4,
        "photos": Array [
          "img/1.png",
          "img/2.png",
        ],
        "picture": "img/1.png",
        "price": 120,
        "rate": 4.8,
        "title": "Beautiful & luxurious studio at great location",
        "type": "apartment",
      }
    }
    onCardMouseEnter={[MockFunction]}
  />
</div>
`;
